File: soluzioni/1/easy2/main.py
elementi = []
with open("input.txt") as fi:
    N = int(fi.readline())
    for _ in range(N):
        a, b = list(map(lambda x: int(x), fi.readline().split()))
        if (a + b) % 2 == 0:
            elementi.append(a + b)

coppie_somma_pari = max(elementi)
with open("output.txt", "w") as fo:
    fo.write(coppie_somma_pari)

File: soluzioni/1/easy2/main.cpp
#include <fstream>

using namespace std;

int main() {
	ifstream input ("input.txt");
	ofstream out ("output.txt");

	int N;
	input  >> N;
	int MAX = -1;

	for (int i = 0; i < N; i++) {
		int a, b, s;
		input >> a >> b;
		s = a + b;
		if (s % 2 == 0 && MAX < s)
			MAX = s;
	}
	
	out << MAX;

	input.close();
	out.close();
	return 0;
}